packages/ra-ui-materialui/src/detail/TabbedShowLayout.js

```diff
@@ -90,33 +90,50 @@ export class TabbedShowLayout extends Component {
             >
                 <Tabs
                     scrollable
-                    value={this.state.value}
+                    value={location.pathname}
                     onChange={this.handleChange}
                     indicatorColor="primary"
                 >
-                    {Children.map(
-                        children,
-                        (tab, index) =>
-                            tab &&
-                            cloneElement(tab, {
-                                context: 'header',
-                                value: index,
-                            })
-                    )}
+                    {Children.map(children, (tab, index) => {
+                        if (!tab) return null;
+                        const tabPath = `${match.url}${tab.props.path
```

Comment:
I'd prefer that you leave that rendering part in `Tab`, in another rendering *context*, as it was done previously. *It's always bad decide too many things for your children* (you can quote me on this).

Reply:
I'm -1 on this. First it's kind of a hack. Second, our two tabbed components manage their tabs differently. I'd rather have them do the same thing as it is easier to reason about them this way.

Finally, we are still deciding things for our children with this hack imo

Reply:
Except it allows to change the way a tab header or content is rendered just by providing a custom `<Tab>` component. With your approach, it's no longer possible. I prefer that we keep the rendering context.

Reply:
Fair enough. Shouldn't we apply the recipe to the FormTab then ?

However, instead of supplying a custom Tab, shouldn't we suggest a custom Form component altogether when further customization is needed?

Reply:
Both the `Tab` and the `FormTab` should use this double render trick.

Writing a custom Form component requires way more knowledge (redux, redux-form, form name, initial values, etc) than writing a custom tab IMO.

Reply:
And when I think about it, it makes `SimpleForm` and `TabbedForm` good candidates for a refactoring based on a `FormController` with a render prop.

Reply:
I was thinking about it too :) However we will probably have to come up with another solution for detecting tabs with errors though

Reply:
Let's merge this and start a refactoring in another PR ?

Reply:
Or should I first revert the Tab change and apply the context pattern to the FormTab?

Reply:
Yes, please readd context, and we’ll refactor the controller part in another PR. 

I’m waiting for this (#1938) PR merge to release 2.1.0.